### DISCUSSION TITLE
Communication with AWS IoT core via MQTT

### DISCUSSION THREAD

**Discussion by Mahdi89** [AxisCommunications/discussions#106] [ACAP] [answered]
Hi there

What would be the best way to enable AWS IoT core to be able to receive events from AXIS cameras via MQTT ? 
It seems that there are .pem and .key files are provided but we cannot see a way to input these files on camera side to enable the connect.

Thanks  

![Screenshot from 2022-11-01 15-46-55](https://user-images.githubusercontent.com/1496173/199278782-b65aad6a-4708-44a5-b062-d0d8becdc3f9.png)

![Screenshot from 2022-11-01 16-01-26](https://user-images.githubusercontent.com/1496173/199279692-ad1ad221-e6e7-4a5a-8f12-778a9bec5d1f.png)

**Comment by mattias-kindborg-at-work** [comment, marked as answer]
Hi @Mahdi89. Well funny that you asked 😄 We've got a [repository with examples targeting AWS](https://github.com/AxisCommunications/acap-integration-examples-aws). One if the examples is showcasing what you are asking for, to connect the Axis device to AWS IoT Core.

We've also got repositories with examples targeting [Azure](https://github.com/AxisCommunications/acap-integration-examples-azure) and [Google Cloud Platform (GCP)](https://github.com/AxisCommunications/acap-integration-examples-gcp).

**Reply by Mahdi89** [reply]
@mattias-kindborg-at-work 
I just noticed in the system logs mqtt is trying connect over TCP whilst I have specified SSL in the setting, do you think that's the issue? Thanks.

![Screenshot from 2022-11-06 14-21-07](https://user-images.githubusercontent.com/1496173/200176333-0be7eab5-85fe-4dea-a344-1dc597ed6155.png)

**Reply by Mahdi89** [reply]
OK - Ading a policy wildcard which allowed publishing into 'any' topic resolved the issue and the camera is now connected to the cloud. 
I suppose the issue was the topic setting and using prefix with topics. Thanks so much.
